app/db/crud/helpers/validate.py

```diff
@@ -7,16 +8,23 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def validate_corpora_ids(db: Session, allowed_corpora_ids: list[str]) -> bool:
-    """Validate whether all given corpus IDs exist in the DB.
+def validate_corpora_ids(
+    db: Session, corpora_ids: list[str], allowed_corpora_ids: Optional[list[str]] = None
+) -> bool:
+    """Validate all given corpus IDs against a list of allowed corpora.
```

Comment:
Maybe this is more accurate as "Validate the allowed list of corpora exist in the database." ?

Reply:
It won't be in the next PR - we'll use this to validate the corpora IDs in the token against the corpora IDs in the database & to validate the corpora IDs provided in the search params against the corpora IDs in the token (if provided)